astroquery/simbad/core.py

```diff
@@ -300,6 +335,41 @@ class SimbadClass(SimbadBaseQuery):
     def __init__(self):
         super().__init__()
         self._VOTABLE_FIELDS = self._VOTABLE_FIELDS.copy()
+        self._mirror = conf.server
+        self._tap = None
+
+    @property
+    def simbad_mirrors(self):
+        """Set of the two Simbad mirrors domains."""
+        return {'simbad.cds.unistra.fr', 'simbad.harvard.edu'}
```

Comment:
We have this in the config already. If it all being moved into class properties then I suppose it should be cleanup from the config. (Though keeping the URLs in the config, rather than in the core.py is more consistent, but no strong preference)

Reply:
Ah yes, this is actually because I was unable to get the other server from there (see the question here https://github.com/astropy/astroquery/issues/2291#issuecomment-1755445731). How do we access the second value of server from the config?

Reply:
I saw that but the error that is raised does not help much someone who does not know beforehand which servers are available

```python
from astroquery.alma import conf
conf.archive_url = "anything"
```

returns

```
TypeError: Provided value for configuration item archive_url not valid: the value "anything" is unacceptable.
```

While with this way of using properties, we'd have 

```python
from astroquery.simbad import Simbad
Simbad.mirror = "anything"
```

which returns

```
ValueError: 'anything' does not correspond to a Simbad mirror, the two existing ones are {'simbad.harvard.edu', 'simbad.cds.unistra.fr'}.
```

this can help a bit more (for example I'm always forgetting that it it simbad.cds.unistra.fr and not simbad.unistra.fr)

(and reading the two versions now makes me think that it might be better with a property name like Simbad.url or Simbad.server. What do you think?)

Reply:
I *think* there is a way to access the list. I mean, the config has to have it somewhere for validation. I'll dig later after dropping kid at school.

Reply:
> I saw that but the error that is raised does not help much someone who does not know beforehand which servers are available

Thus there is the list added in alma outside of the config. But of course something nicer, directly from the config would be the best.

Reply:
It is a little convoluted but the info is there. Here is an example from core lib `astropy`:

```python
>>> from astropy.utils.iers import conf
>>> c = conf.__class__.__dict__["iers_degraded_accuracy"]  # Grabs the ConfigItem instance
>>> c.cfgtype  # Valid options encoded in the return string
'option(error, warn, ignore)'
```

Here is how it is defined at a high level: https://github.com/astropy/astropy/blob/1ae26ae0fcf5b8ffe0563d73be7787e9c835bf0e/astropy/utils/iers/iers.py#L190-L195

```python
class Conf(_config.ConfigNamespace):
    iers_degraded_accuracy = _config.ConfigItem(
        ["error", "warn", "ignore"],
        "IERS behavior if the range of available IERS data does not "
        "cover the times when converting time scales, potentially leading "
        "to degraded accuracy.",
    )
```

The implementation predates me and a lot of the internals are handed off to the external configobj package that astropy bundles. So don't ask me why it has to be like this. Hope this helps!

Reply:
It does! Thanks a lot.

Reply:
yes, I think this alma solution sounds reasonable, and way more readable and understandable than the suggested workaround. I suspect the whole config system has a lot of historical debt as being an old and never really touched up part of the package.

Reply:
Yes, that is definitely cleaner than accessing config in weird places. 👏